runtime/etc/src/mli/client_runtime.c

```diff
@@ -81,30 +95,50 @@ void chpl_library_init(int argc, char** argv) {
 
   // Just hardcode these values for now.
   const char* ip = "localhost";
-  const char* setupPort = "5554";
-  const char* mainport = "5555";
-  const char* argport = "5556";
-  const char* resport = "5557";
 
   if (initialized) { return; }
   initialized = 1;
 
   // Set up the clientside ZMQ sockets.
   chpl_mli_client_init(&chpl_client);
 
-  chpl_mli_bind(chpl_client.setup_sock, ip, setupPort);
+  chpl_mli_bind(chpl_client.setup_sock, ip, "*");
 
   char* setup_sock_conn = chpl_mli_connection_info(chpl_client.setup_sock);
   chpl_mli_debugf("Setup socket used: %s\n", setup_sock_conn);
 
-  chpl_mli_client_launch(argc, argv);
-
+  // Send the setup socket as the last argument when launching the server.
+  int argcPlusSock = argc + 2;
+  char** argvPlusSock = mli_malloc(argcPlusSock * sizeof(char*));
+  for (int i = 0; i < argc; i++) {
+    chpl_mli_debugf("passing along arg %d: %s\n", i, argv[i]);
+    argvPlusSock[i] = argv[i];
+  }
+  argvPlusSock[argc] = "--chpl-mli-socket-loc";
+  argvPlusSock[argc + 1] = setup_sock_conn;
+  chpl_mli_debugf("spawning server with %d args\n", argcPlusSock);
```

Comment:
Capitalize "passing"? I don't know if we care about proper capitalization and stuff for debug messages or not, since realistically we're the only ones who are ever going to see them. 

I normally don't care too much about formatting, but #13030 has me up in arms...